Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -0,0 +1,256 @@
+
+                // Object migrationObject is a scene Game Object
+                if (String.IsNullOrEmpty(assetPath))
+                {
+                    MigrateGameObjectHierarchy((GameObject)migrationObject);
+                }
+                else
+                {
+                    if (migrationObject is GameObject)
+                    {
+                        MigratePrefab(assetPath);
+                    }
+                    else if (migrationObject is SceneAsset)
+                    {
+                        MigrateScene(assetPath);
+                    }
+                }
+            }
+            migrationObjects.Clear();
+
+            if (!String.IsNullOrEmpty(previousScenePath) && previousScenePath != EditorSceneManager.GetActiveScene().path)
+            {
+                EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
+            }
+            return true;
+        }
+
+        private void MigrateScene(String path)
+        {
+            if (!AssetDatabase.LoadAssetAtPath(path, typeof(SceneAsset)))
+            {
+                return;
+            }
+            Scene scene = EditorSceneManager.OpenScene(path);
+
+            foreach (var parent in scene.GetRootGameObjects())
+            {
+                MigrateGameObjectHierarchy(parent);
+            }
+            EditorSceneManager.SaveScene(scene);
+        }
+
+        private void MigratePrefab(String path)
+        {
+            if (!AssetDatabase.LoadAssetAtPath(path, typeof(GameObject)))
+            {
+                return;
+            }
+            var parent = UnityEditor.PrefabUtility.LoadPrefabContents(path);
+
+            MigrateGameObjectHierarchy(parent);
+
+            UnityEditor.PrefabUtility.SaveAsPrefabAsset(parent, path);
+            PrefabUtility.UnloadPrefabContents(parent);
+        }
+
+        private void MigrateGameObjectHierarchy(GameObject parent)
+        {
+            if (!parent)
+            {
+                return;
+            }
+
+            foreach (var child in parent.GetComponentsInChildren<Transform>())
+            {
+                try
+                {
+                    if (migrationHandlerInstance.CanMigrate(child.gameObject))
+                    {
+                        migrationHandlerInstance.Migrate(child.gameObject);
+                        Debug.Log($"Successfully migrated {parent.name} object");
+                    }
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError($"{e.Message}: GameObject {parent.name} could not be Migrated");
+                }
+            }
+        }
+
+        private static List<string> FindAllAssetsOfType(Type type)
+        {
+            return AssetDatabase.GetAllAssetPaths()
+                .Select(x => x)
+                .Where(x => AssetDatabase.GetMainAssetTypeAtPath(x) == type)
+                .ToList();
+        }
```

Comment:
You may be able to just use AssetDatbase.Find("t:{type1} {type2}") also consider making the function 
`private static List<string> FindAllAssetsOfTypes(Type[] types)`
https://answers.unity.com/questions/486545/getting-all-assets-of-the-specified-type.html

If that AssetDatabase.Find() doesn't work, then consider at least doing the Type array like below. This way you aren't pulling ALL assets everytime and then filtering. 

```
        private static List<string> FindAllAssetsOfTypes(Type[] types) // or hashset
        {
            return AssetDatabase.GetAllAssetPaths()
                //.Select(x => x) - not necessary?
                .Where(x => types.Contains(AssetDatabase.GetMainAssetTypeAtPath(x)))
                .ToList();
        }
```

```

Reply:
Changed that. It is indeed exactely 10 times faster (30 vs 3 ms)